check_python.sh

```diff
@@ -111,50 +103,17 @@ pushd "${PROJECT_REPO}" > /dev/null || exit 1
 
     IFS=" " read -r -a pylint_cmd <<< "${pylint} ${pyl_opts} ${script_file}"
     IFS=" " read -r -a pylint3_cmd <<< "${pylint} ${pyl3_opts} ${script_file}"
-    if [ ! -f "${script_file}" ]; then
-        continue
-    fi
-    if [[ ${script_file} == *.py ]]; then
-      # if there is a shebang use it.
-      if (head -1 "$script_file" | grep -q -E '^#!(/usr)?/bin/.*python'); then
-        if (head -1 "$script_file" | \
-            grep -q -E '^#!(/usr)?/bin/.*python3'); then
-          if ! python3 "${pylint3_cmd[@]}" --msg-template "${tmpl}" >> \
-            "${PYLINT_OUT}" 2>&1; then
-            (( rc=rc+PIPESTATUS[0] ))
-          fi
-        else
-          if ! python "${pylint_cmd[@]}" --msg-template "${tmpl}" >> \
-            "${PYLINT_OUT}" 2>&1; then
-            (( rc=rc+PIPESTATUS[0] ))
-          fi
+    if file "$script_file" | grep -q -E 'Python script'; then
+      if head -1 "$script_file" | grep -q -e 'python3'; then
+        if ! python3 "${pylint3_cmd[@]}" --msg-template "${tmpl}" >> \
+          "${PYLINT_OUT}" 2>&1; then
+          (( rc=rc+PIPESTATUS[0] ))
         fi
       else
-        if [ "${def_python}" == "python" ]; then
-          if ! python "${pylint_cmd[@]}" --msg-template "${tmpl}" >> \
-            "${PYLINT_OUT}" 2>&1; then
-            (( rc=rc+PIPESTATUS[0] ))
-          fi
-        else
-          if ! python3 "${pylint3_cmd[@]}" --msg-template "${tmpl}" >> \
-            "${PYLINT_OUT}" 2>&1; then
-            (( rc=rc+PIPESTATUS[0] ))
-          fi
-        fi
-      fi
-    else
-      if (head -1 "$script_file" | grep -q -E '^#!(/usr)?/bin/.*python'); then
-        if (head -1 "$script_file" | \
-            grep -q -E '^#!(/usr)?/bin/.*python3'); then
-          if ! python3 "${pylint3_cmd[@]}" --msg-template "${tmpl}" >> \
-            "${PYLINT_OUT}" 2>&1; then
-            (( rc=rc+PIPESTATUS[0] ))
-          fi
-        else
-          if ! python "${pylint_cmd[@]}" --msg-template "${tmpl}" >> \
-            "${PYLINT_OUT}" 2>&1; then
-            (( rc=rc+PIPESTATUS[0] ))
-          fi
+        # Default python, which on current Fedora, etc is same as python3
```

Comment:
It was previously checking to see what the default python was and using it if the shebang just said python, otherwise it did a python2 lint for just plain python.

Now for "python" it uses the distro default python, which on Centos is python2 and Fedora is python3.

For python3 it uses python3.

It does not lint the python files a second time now or before.

I removed most of the logic, but since on daos for the moment we have not tested this code with the stage running CentOS, I left the separate lints for python and python3 in there.

Once we convert the daos checkpatch step to use a fedora docker file, then we can remove the python3 checks.